webpack/ForemanTasks/Components/TasksTable/TasksTableActions.js

```diff
@@ -171,28 +186,55 @@ export const bulkCancel = ({
   url,
   parentTaskID,
 }) => async dispatch => {
-  let notAllActionable = false;
-  let someActionable = false;
-  const promises = selected.map(task => {
-    if (task.isCancellable) {
-      someActionable = true;
-      return dispatch(cancelTaskRequest(task.id, task.name));
-    }
-    notAllActionable = true;
-    return null;
-  });
-
-  if (notAllActionable)
+  const cancelTasks = selected.filter(task => task.isCancellable);
+  if (cancelTasks.length < selected.length)
     dispatch(
       addToast({
-        type: 'warning',
-        message: __('Not all the selected tasks can be canceled'),
+        type: TOAST_TYPES.WARNING,
+        message: __('Not all the selected tasks can be cancelled'),
       })
     );
-  if (someActionable) {
-    await Promise.all(promises);
-    dispatch(getTableItems(url));
-    dispatch(fetchTasksSummary(getURIQuery(url).time, parentTaskID));
+  if (cancelTasks.length) {
+    try {
+      dispatch({ type: TASKS_CANCEL_REQUEST });
+      const { data } = await bulkCancelRequest(cancelTasks);
+      dispatch({ type: TASKS_CANCEL_SUCCESS });
+      const toastInfo = {
+        cancelled: { type: TOAST_TYPES.SUCCESS, text: 'was cancelled' },
+        skipped: {
+          type: TOAST_TYPES.WARNING,
+          text: 'task has to be cancellable',
+        },
+      };
+      const toastDispatch = (type, name) =>
+        dispatch(
+          addToast({
+            type: toastInfo[type].type,
+            message: sprintf(__('%(name)s Task execution %(type)s'), {
+              name,
+              type: toastInfo[type].text,
+            }),
+          })
+        );
+      ['cancelled', 'skipped'].forEach(type => {
+        data[type] &&
+          data[type].forEach(task => {
+            toastDispatch(type, task.action);
+          });
+      });
+      if (data.cancelled) {
+        dispatch(getTableItems(url));
+        dispatch(fetchTasksSummary(getURIQuery(url).time, parentTaskID));
+      }
+    } catch (error) {
+      dispatch({ type: TASKS_CANCEL_FAILURE, error });
+      dispatch(
+        addToast({
+          type: TOAST_TYPES.ERROR,
+          message: `${__(`Cannot cancel tasks at the moment`)} ${error}`,
+        })
+      );
+    }
```

Comment:
Can you add a test for the `TASKS_CANCEL_FAILURE` ?

Reply:
added!